apex/contrib/optimizers/distributed_fused_adam.py

```diff
@@ -1,26 +1,84 @@
 import collections
 import contextlib
 import enum
-import importlib
 import inspect
 import io
-import math
+import itertools
 import threading
 
 import torch
-import amp_C
-from apex.multi_tensor_apply import multi_tensor_applier
 from torch.distributed.distributed_c10d import _get_default_group, _get_global_rank
+from apex.multi_tensor_apply import multi_tensor_applier
+import amp_C
+import distributed_adam_cuda
+
+_FOUND_DEPRECATED_FUSED_ADAM = False
+try:
+    import fused_adam_cuda
+    _FOUND_DEPRECATED_FUSED_ADAM = True
+except ImportError:
+    import warnings
+    warnings.warn(
+        'Could not find recommended CUDA kernels when importing '
+        'apex.contrib.optimizers.distributed_fused_adam. '
+        'Apex should be installed with the following options: '
+        '`--cpp_exp`, `--cuda_ext`, '
+        '`--distributed_adam`, `--deprecated_fused_adam`.'
+    )
```

Comment:
does this optimizer need `fused_adam_cuda`? I don't think this try-except raises when the extension is not found

Reply:
`fused_adam_cuda` has a fused copy kernel that we use here:

https://github.com/timmoon10/apex/blob/2d5ebb793ca212773bc0af4b15afaee2cea5e82f/apex/contrib/optimizers/distributed_fused_adam.py#L66

If we can't find that kernel, then we fall back to performing the copies one at a time.

Reply:
I see, nit-picking though, "APEX should be installed with the following option" sounded a little bit misleading to me so how about clarifying this optimizer will be faster with `--deprecated_fused_adam`?